packages/edit-post/src/components/preferences-modal/index.js

```diff
@@ -213,10 +208,7 @@ export default function EditPostPreferencesModal() {
 							<PostTaxonomies
 								taxonomyWrapper={ ( content, taxonomy ) => (
 									<EnablePanelOption
-										label={ get( taxonomy, [
-											'labels',
-											'menu_name',
-										] ) }
+										label={ taxonomy.labels.menu_name }
```

Comment:
We're already using `taxonomy` as a non-nullish object here, and a valid `taxonomy` entity record will always have a `labels` object.